waspc/data/Generator/templates/sdk/wasp/client/operations/queries/core.ts

```diff
@@ -13,7 +13,9 @@ export function createQuery<BackendQuery extends GenericBackendQuery>(
 ): QueryFor<BackendQuery> {
   const queryRoute = makeOperationRoute(relativeQueryPath)
 
-  async function query(queryKey, queryArgs) {
+  async function query(queryArgs) {
+    // Assumes `addMetadataToQuery` added the `queryCacheKey` property to the query.
+    const queryKey = (query as any).queryCacheKey
```

Comment:
Todo: fix the issue in `operations/core.ts#useQuery` where they call it with the cache key.

Test for queries that have a payload.